File: Edu-JavaScript/src/com/jetbrains/edu/javascript/learning/checkio/connectors/JsCheckiOOAuthConnector.java
package com.jetbrains.edu.javascript.learning.checkio.connectors;

import com.jetbrains.edu.javascript.learning.checkio.JsCheckiOSettings;
import com.jetbrains.edu.javascript.learning.checkio.utils.JsCheckiONames;
import com.jetbrains.edu.javascript.learning.checkio.utils.JsCheckiOOAuthBundle;
import com.jetbrains.edu.learning.checkio.account.CheckiOAccount;
import com.jetbrains.edu.learning.checkio.connectors.CheckiOOAuthConnector;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class JsCheckiOOAuthConnector extends CheckiOOAuthConnector {
  private static final String CLIENT_ID = JsCheckiOOAuthBundle.messageOrDefault("jsCheckioClientId", "");
  private static final String CLIENT_SECRET = JsCheckiOOAuthBundle.messageOrDefault("jsCheckioClientSecret", "");

  protected JsCheckiOOAuthConnector() {
    super(CLIENT_ID, CLIENT_SECRET);
  }

  @Nullable
  @Override
  public CheckiOAccount getAccount() {
    return JsCheckiOSettings.getInstance().getAccount();
  }

  @Override
  public void setAccount(@Nullable CheckiOAccount account) {
    JsCheckiOSettings.getInstance().setAccount(account);
  }

  @NotNull
  @Override
  protected String getOAuthServicePath() {
    return JsCheckiONames.JS_CHECKIO_OAUTH_SERVICE_PATH;
  }

  @NotNull
  @Override
  protected String getPlatformName() {
    return JsCheckiONames.JS_CHECKIO;
  }


  private static class Holder {
    private static final JsCheckiOOAuthConnector INSTANCE = new JsCheckiOOAuthConnector();
  }

  public static JsCheckiOOAuthConnector getInstance() {
    return Holder.INSTANCE;
  }
}

File: Edu-JavaScript/src/com/jetbrains/edu/javascript/learning/checkio/connectors/JsCheckiOOAuthConnector.kt
package com.jetbrains.edu.javascript.learning.checkio.connectors

import com.jetbrains.edu.javascript.learning.checkio.JsCheckiOSettings
import com.jetbrains.edu.javascript.learning.checkio.utils.JsCheckiOOAuthBundle
import com.jetbrains.edu.learning.checkio.account.CheckiOAccount
import com.jetbrains.edu.learning.checkio.connectors.CheckiOOAuthConnector
import com.jetbrains.edu.learning.checkio.utils.CheckiONames
import com.jetbrains.edu.learning.checkio.utils.CheckiONames.JS_CHECKIO_PREFIX

object JsCheckiOOAuthConnector : CheckiOOAuthConnector() {
  override var account: CheckiOAccount?
    get() = JsCheckiOSettings.getInstance().account
    set(account) {
      JsCheckiOSettings.getInstance().account = account
    }

  override val clientId: String = JsCheckiOOAuthBundle.value("jsCheckioClientId")

  override val clientSecret: String = JsCheckiOOAuthBundle.value("jsCheckioClientSecret")

  override val serviceName: String = getCheckiOServiceName(JS_CHECKIO_PREFIX.toLowerCase())

  override val platformName: String = CheckiONames.JS_CHECKIO
}